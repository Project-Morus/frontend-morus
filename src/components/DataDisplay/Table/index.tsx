import { TBodySC, TableSC, TdSC, TrSC, WrapperMessageSC } from "./styles";
import { ITableProps } from "./types";

function Table<T extends Record<string, any>>({ headerCells, rowCells, emptyMessage, ...props }: ITableProps<T>) {
  return (
    (rowCells?.length) ? (
      <TableSC {...props}>
        <thead>
          <tr>
            {headerCells.map(({ colName }, index) => (
              <th key={index}>{colName}</th>
            ))}
          </tr>
        </thead>

        <TBodySC>
          {rowCells.map((item, rowIndex) => (
            <TrSC key={rowIndex}>
              {Object.keys(item).map((key, columnIndex) => (
                <TdSC key={columnIndex}>{item[key]}</TdSC>
              ))}
            </TrSC>
          ))}
        </TBodySC>
      </TableSC>
    ) : <WrapperMessageSC>{emptyMessage}</WrapperMessageSC>
  )
}


export default Table;
