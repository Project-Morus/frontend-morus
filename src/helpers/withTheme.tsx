import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

