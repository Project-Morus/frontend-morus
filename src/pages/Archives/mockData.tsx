import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";
import Icons from "./parts/Icons";


export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Tamanho" },
  { colName: "Nome" },
  { colName: "Data da postagem" },
  { colName: "Ações" },
];

export const ROW_NAME_CELLS = [
  { size: "35MB", name: "Regimento Interno do Condomínio Ilha de Capri", created_at: "11/05/2023", actions: <Icons /> },
  { size: "15MB", name: "Convenção do condomínio Ilha de Capri", created_at: "12/04/2023", actions: <Icons /> },
  { size: "2MB", name: "Edital de Convocação para Assembleia Geral Ordinária", created_at: "10/04/2023", actions: <Icons /> },
  { size: "7MB", name: "Alvará de construção da prefeitura de Vila Velha-ES", created_at: "12/02/2022", actions: <Icons /> },
  { size: "101MB", name: "Plantas e Projetos", created_at: "19/01/2022", actions: <Icons /> },
];
