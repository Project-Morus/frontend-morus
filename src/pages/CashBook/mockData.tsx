import Icon from "../../components/Icon";
import { ITableHeaderProps } from "../../components/Table/types";

const IconsSC = () => {
  return (
    <div style={{ display: "flex", gap: "0.8rem" }}>
      <Icon name="ph-pencil-simple-line" size="20" />
      <Icon name="ph-trash" color="#CE323B" size="20" />
    </div>
  );
};

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Torre" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Data" },
  { colName: "Categoria" },
  { colName: "Ações" },
];

export const ROW_NAME_CELLS = [
  {
    tower: "Todos",
    description: "Salário do porteiro",
    value: "-R$3.250,50",
    date: "05/02/2024",
    category: "Funcionário",
    actions: <IconsSC />,
  },
  {
    tower: "Norte",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Sul",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Sul",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
  {
    tower: "Norte",
    description: "Taxa de condomínio",
    value: "R$1.200,50",
    date: "05/02/2024",
    category: "Taxa de condomínio",
    actions: <IconsSC />,
  },
];
