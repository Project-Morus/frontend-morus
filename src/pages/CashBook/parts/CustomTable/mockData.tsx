import { ITableHeaderProps } from "../../types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Torre" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Data" },
  { colName: "Categoria" },
  { colName: "Ações" },
];
