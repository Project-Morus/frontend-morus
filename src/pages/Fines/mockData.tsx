import { ITableHeaderProps } from "../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Nome" },
  { colName: "Motivo" },
  { colName: "Valor" },
  { colName: "Juros" },
  { colName: "Data" },
  { colName: "Vencimento" },
  { colName: "Status" },
  { colName: "Ações" },
];
