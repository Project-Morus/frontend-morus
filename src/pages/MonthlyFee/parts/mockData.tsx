import { ITableHeaderProps } from "../../../components/DataDisplay/Table/types";

export const HEADER_TABLE_CELLS: ITableHeaderProps[] = [
  { colName: "Nome" },
  { colName: "Descrição" },
  { colName: "Valor" },
  { colName: "Vencimento" },
  { colName: "Status" },
  { colName: "Ações" },
];
