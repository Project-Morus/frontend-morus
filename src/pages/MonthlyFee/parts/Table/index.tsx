import { useTheme } from "styled-components";
import { NewTable, Icon } from "../../../../components";
import { formatDate } from "../../../../helpers/date";
import { HEADER_TABLE_CELLS } from "../mockData";
import Status from "../Status";
import { WrapperIcons } from "./styles";

function TableMonthlyFee() {
  const theme = useTheme();

  // if (isLoading)
  //   return (
  //     <ContentLoaderSC>
  //       <Loader />
  //     </ContentLoaderSC>
  //   );

  return (
    <>
      <NewTable.Container>
        <NewTable.Head>
          <NewTable.Row>
            {HEADER_TABLE_CELLS.map((header, index) => (
              <NewTable.CellHeader key={index}>
                {header.colName}
              </NewTable.CellHeader>
            ))}
          </NewTable.Row>
        </NewTable.Head>

        <NewTable.Body>
          <NewTable.Row>
            <NewTable.Cell>Taxa de Condominio</NewTable.Cell>
            <NewTable.Cell>Alguma coisa</NewTable.Cell>
            <NewTable.Cell>R$1.200,50</NewTable.Cell>
            <NewTable.Cell>01/11/2023</NewTable.Cell>
            <NewTable.Cell>
              <Status recurring={false} />
            </NewTable.Cell>
            <NewTable.Cell>
              <WrapperIcons>
                <Icon name="ph-pencil-simple-line" />
                <Icon name="ph-trash" color={theme.colors.red[500]} />
              </WrapperIcons>
            </NewTable.Cell>
          </NewTable.Row>
        </NewTable.Body>
      </NewTable.Container>
    </>
  );
}

export default TableMonthlyFee;
