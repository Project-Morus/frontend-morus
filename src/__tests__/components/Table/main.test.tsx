import { render, screen } from '@testing-library/react'
import { withTheme } from '../../../helpers/withTheme'
import Table from '../../../components/Table'
import { HEADER_TABLE_CELLS, ROW_NAME_CELLS } from './mockData'

const TableWithTheme = withTheme(Table)

describe('Table component', () => {
  test('Verify if component is rendering', () => {
    render(<TableWithTheme headerCells={HEADER_TABLE_CELLS} rowCells={ROW_NAME_CELLS} />)

    expect(screen.getByRole('table'))
  })

  test('Verify if emptyMessage is rendering', () => {
    render(<TableWithTheme headerCells={HEADER_TABLE_CELLS} rowCells={[]} emptyMessage="Vazio!" />)

    expect(screen.findByText('Vazio!'))
  })


  test('Snapshot Table', () => {
    const { asFragment } = render(
      <TableWithTheme headerCells={HEADER_TABLE_CELLS} rowCells={ROW_NAME_CELLS} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
