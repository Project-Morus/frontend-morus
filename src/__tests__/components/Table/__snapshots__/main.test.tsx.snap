// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component Snapshot Table 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  border-spacing: 0 0.75rem;
  border-radius: 5px;
  background-color: #F8F8F8;
  padding: 0.75rem;
}

.c0 thead {
  text-align: left;
  color: #999999;
}

.c0 thead th {
  padding-left: 0.75rem;
}

.c3 {
  padding: 1rem;
  font-size: 1rem;
  font-weight: 700;
  color: #4E4E4E;
}

.c1 {
  background-color: #FFF;
}

.c2 {
  box-shadow: 0px 4px 4px 0px rgba(0, 74, 173, 0.15);
  border-radius: 5px;
}

.c4 {
  color: #0353A4;
}

.c6 {
  color: #CE323B;
}

.c5 {
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  width: fit-content;
  width: -moz-fit-content;
  height: fit-content;
  height: -moz-fit-content;
}

<table
    class="c0"
  >
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Nome
        </th>
        <th>
          Data
        </th>
        <th>
          Status
        </th>
        <th>
          Ações
        </th>
      </tr>
    </thead>
    <tbody
      class="c1"
    >
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          1
        </td>
        <td
          class="c3"
        >
          Eduardo Simon
        </td>
        <td
          class="c3"
        >
          20/12/2023
        </td>
        <td
          class="c3"
        >
          Em progresso
        </td>
        <td
          class="c3"
        >
          <div
            style="display: flex; gap: 0.8rem;"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
                data-testid="ph-pencil-simple-line"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M227.32,73.37,182.63,28.69a16,16,0,0,0-22.63,0L36.69,152A15.86,15.86,0,0,0,32,163.31V208a16,16,0,0,0,16,16H216a8,8,0,0,0,0-16H115.32l112-112A16,16,0,0,0,227.32,73.37ZM92.69,208H48V163.31l88-88L180.69,120ZM192,108.69,147.32,64l24-24L216,84.69Z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c5"
                data-testid="ph-trash"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          2
        </td>
        <td
          class="c3"
        >
          Eduardo Simon
        </td>
        <td
          class="c3"
        >
          20/12/2023
        </td>
        <td
          class="c3"
        >
          Em progresso
        </td>
        <td
          class="c3"
        >
          <div
            style="display: flex; gap: 0.8rem;"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
                data-testid="ph-pencil-simple-line"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M227.32,73.37,182.63,28.69a16,16,0,0,0-22.63,0L36.69,152A15.86,15.86,0,0,0,32,163.31V208a16,16,0,0,0,16,16H216a8,8,0,0,0,0-16H115.32l112-112A16,16,0,0,0,227.32,73.37ZM92.69,208H48V163.31l88-88L180.69,120ZM192,108.69,147.32,64l24-24L216,84.69Z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c5"
                data-testid="ph-trash"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          3
        </td>
        <td
          class="c3"
        >
          Eduardo Simon
        </td>
        <td
          class="c3"
        >
          20/12/2023
        </td>
        <td
          class="c3"
        >
          Em progresso
        </td>
        <td
          class="c3"
        >
          <div
            style="display: flex; gap: 0.8rem;"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
                data-testid="ph-pencil-simple-line"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M227.32,73.37,182.63,28.69a16,16,0,0,0-22.63,0L36.69,152A15.86,15.86,0,0,0,32,163.31V208a16,16,0,0,0,16,16H216a8,8,0,0,0,0-16H115.32l112-112A16,16,0,0,0,227.32,73.37ZM92.69,208H48V163.31l88-88L180.69,120ZM192,108.69,147.32,64l24-24L216,84.69Z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c5"
                data-testid="ph-trash"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          4
        </td>
        <td
          class="c3"
        >
          Eduardo Simon
        </td>
        <td
          class="c3"
        >
          20/12/2023
        </td>
        <td
          class="c3"
        >
          Em progresso
        </td>
        <td
          class="c3"
        >
          <div
            style="display: flex; gap: 0.8rem;"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
                data-testid="ph-pencil-simple-line"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M227.32,73.37,182.63,28.69a16,16,0,0,0-22.63,0L36.69,152A15.86,15.86,0,0,0,32,163.31V208a16,16,0,0,0,16,16H216a8,8,0,0,0,0-16H115.32l112-112A16,16,0,0,0,227.32,73.37ZM92.69,208H48V163.31l88-88L180.69,120ZM192,108.69,147.32,64l24-24L216,84.69Z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c5"
                data-testid="ph-trash"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="c2"
      >
        <td
          class="c3"
        >
          5
        </td>
        <td
          class="c3"
        >
          Eduardo Simon
        </td>
        <td
          class="c3"
        >
          20/12/2023
        </td>
        <td
          class="c3"
        >
          Em progresso
        </td>
        <td
          class="c3"
        >
          <div
            style="display: flex; gap: 0.8rem;"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
                data-testid="ph-pencil-simple-line"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M227.32,73.37,182.63,28.69a16,16,0,0,0-22.63,0L36.69,152A15.86,15.86,0,0,0,32,163.31V208a16,16,0,0,0,16,16H216a8,8,0,0,0,0-16H115.32l112-112A16,16,0,0,0,227.32,73.37ZM92.69,208H48V163.31l88-88L180.69,120ZM192,108.69,147.32,64l24-24L216,84.69Z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="c6"
            >
              <div
                class="c5"
                data-testid="ph-trash"
              >
                <svg
                  fill="currentColor"
                  height="20"
                  viewBox="0 0 256 256"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
